### PR TITLE
UDP Concepts "guide" created as companion piece to UDP commands in Command Ref

### PR DESCRIPTION
**Pull request by jrhodie**

### CODE REVIEW

File: site/en/reference/cli/concepts/udp.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
+# Send UDP Messages with OT CLI
+
+Thread networks offer several `udp` commands for establishing peer-to-peer communication
+using UDP sockets. Each peer needs its own socket.
+
+In addition to opening the required sockets, one requirement is that the socket
+on the receiving end of the initial communication be bound.
+
+The examples that follow demonstrate how you can open and bind sockets, how to
+to specify a peer with which to associate a local socket, and how to send messages
```
Comment (abtink) [inline site/en/reference/cli/concepts/udp.md:10]:
```suggestion
to connect a socket, and how to send messages
```

File: site/en/reference/cli/concepts/udp.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+close
+connect
+linksecurity
+open
+send
+Done
+
+### `open` command
+
+Use the `udp open` command to open a socket to begin UDP communication.
```
Comment (abtink) [inline site/en/reference/cli/concepts/udp.md:28]: We should not use the term "a socket" as all `cli udp` commands interact with just one specific example socket provided by CLI.

We can refer to it as "the example socket" which make it consistent with the `README_UDP.md`.

File: site/en/reference/cli/concepts/udp.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+### `bind` command
+
+After you `open` a socket, you can run a `udp bind` command to assign an IPv6 address
+and a port to the open socket. This binds the socket for communication. Assigning the
+IPv6 address and port is also referred to as naming the socket. Binding is required to
+initially receive a UDP message. However, binding a socket is not required
+on the node that is sending the first message. If the initiating node opens but
+does not bind a socket, the socket will choose its own local IP address and ephemeral port
+when sending the message.
```
Comment (abtink) [inline site/en/reference/cli/concepts/udp.md:39]: Recommend simplifying this:
```suggestion
IPv6 address and port is also referred to as naming the socket. 

If the socket is not already bound, connecting it and/or using it to send will bind it to use an ephemeral port.
```

File: site/en/reference/cli/concepts/udp.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+### `connect` command
+
+A `udp connect` command can be used to establish UDP communication with one specific peer socket
+by specifying the IP address and UDP port number of the peer. This command can later be
```
Comment (abtink) [inline site/en/reference/cli/concepts/udp.md:44]: Suggest to simplify:
`udp connect` can be used to connect the example socket to a peer socket address.

File: site/en/reference/cli/concepts/udp.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+followed by a `udp send` command. While the `udp connect` command is not required in
+the communication between two UDP sockets, it does provide the simplicity of not having to
+specify the destination IP address and port each time a `udp send` command is issued. This
+is because the connected peer socket would be used by default. Issuing the `udp connect`
+command will bind the local socket to an ephemeral port if the local socket has
+not already been bound.
```
Comment (abtink) [inline site/en/reference/cli/concepts/udp.md:50]: Suggest removing extra details.
```suggestion
followed by a `udp send` command to send to send to peer.

If socket is not already bound, issuing the `udp connect` command will bind also bind the socket.
```

File: site/en/reference/cli/concepts/udp.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+is because the connected peer socket would be used by default. Issuing the `udp connect`
+command will bind the local socket to an ephemeral port if the local socket has
+not already been bound. 
+
+### `send` command
+
+A `udp send` commands sends a message to the socket whose IP address and UDP port can
+specified with the command variables. If the IP address and port are not specified in the
+`udp send` command, the message gets sent to the socket that was specified in the `udp connect` command.
+Issuing the `udp send` command will bind the local socket to an ephemeral port
```
Comment (abtink) [inline site/en/reference/cli/concepts/udp.md:57]: Suggest not using "local socket" (all the `cli` commands are interacting with the one specific example socket provided by CLI).

File: site/en/reference/cli/concepts/udp.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+followed by a `udp send` command. While the `udp connect` command is not required in
+the communication between two UDP sockets, it does provide the simplicity of not having to
+specify the destination IP address and port each time a `udp send` command is issued. This
+is because the connected peer socket would be used by default. Issuing the `udp connect`
+command will bind the local socket to an ephemeral port if the local socket has
+not already been bound. 
+
+### `send` command
+
+A `udp send` commands sends a message to the socket whose IP address and UDP port can
```
Comment (abtink) [inline site/en/reference/cli/concepts/udp.md]: "sends a message to the socket" is not technically correct. We send a UDP message using the example socket to a destination.

File: site/en/reference/cli/concepts/udp.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
+# Send UDP Messages with OT CLI
+
+Thread networks offer several `udp` commands for establishing peer-to-peer communication
+using UDP sockets. Each peer needs its own socket.
+
+In addition to opening the required sockets, one requirement is that the socket
+on the receiving end of the initial communication be bound.
+
+The examples that follow demonstrate how you can open and bind sockets, how to
```
Comment (abtink) [inline site/en/reference/cli/concepts/udp.md:9]: Useful to mention that `cli udp` interact with one example socket.  

`cli udp` command does not really let user create new sockets .

**Review by abtink — COMMENTED** [review]
Some suggestions below. Thanks.

File: site/en/reference/cli/concepts/udp.md [hunk 9]
```diff
@@ -0,0 +1,185 @@
+# Send UDP Messages with OT CLI
```
Comment (Vyrastas) [inline site/en/reference/cli/concepts/udp.md:1]: Should we update to be more clear how the CLI is intended for use, something like:

`# Test UDP functionality with OT CLI`

**Review by Vyrastas — CHANGES_REQUESTED** [review]

File: site/en/reference/cli/concepts/udp.md [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+if the socket has not already been bound.
+
+### `close` command
+
+It is recommended that you use the `udp close` command to close the socket when
+the socket is no longer needed.
+
+### `linksecurity` command
+
+The `udp linksecurity` command can be used to enable/disable MAC/link level security for messages.
```
[the remark this answers is not in the document]
Comment (jrhodie) [reply site/en/reference/cli/concepts/udp.md:59]: @abtink Can I make Jeff's suggested change here to say "data link level?"

[the remark this answers is not in the document]
Comment (abtink) [reply site/en/reference/cli/concepts/udp.md:59]: yes. we can say enable or disable "data link layer security"

File: site/en/reference/cli/concepts/udp.md (hunk 9 above)
Comment (jrhodie) [reply site/en/reference/cli/concepts/udp.md:1]: I like it; will change to that.  @abtink - agree?

**Review by abtink — APPROVED** [review]
LGTM. Thanks. 👍

File: site/en/reference/cli/concepts/udp.md [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+### `close` command
+
+It is recommended that you use the `udp close` command to close the socket when
+the socket is no longer needed.
+
+### `linksecurity` command
+
+The `udp linksecurity` command can be used to enable or disable data-link layer security for messages. 
+
+## Send a Message Between Two Nodes
```
Comment (Vyrastas) [inline site/en/reference/cli/concepts/udp.md:61]: Headers should be sentence case

Send a message between two nodes

File: site/en/reference/cli/concepts/udp.md [hunk 12]
```diff
@@ -0,0 +1,186 @@
+# Test UDP Functionality With OT CLI
```
Comment (Vyrastas) [inline site/en/reference/cli/concepts/udp.md:1]: Also add this file to https://github.com/openthread/ot-docs/blob/main/site/en/reference/cli/_toc.yaml

File: site/en/reference/cli/concepts/udp.md [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+    For complete options with `udp send`, refer to
+    [udp send](https://openthread.io/reference/cli/commands#udp_send).
+
+1. Node 1 confirms receipt of the message from Node 2:
+
+   ```
+   > 5 bytes from fdde:ad00:beef:0:dac3:6792:e2e:90d8 49153 hello
+   ```
+
+## Connect the Socket to the Peer Socket Address, then Send a Message Between Two Nodes
```
Comment (Vyrastas) [inline site/en/reference/cli/concepts/udp.md:111]: Sentence case header

**Review by Vyrastas — CHANGES_REQUESTED** [review]
Looks good just a few more nits

**Review by Vyrastas — APPROVED** [review]

File: site/en/reference/cli/concepts/udp.md [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+    ```
+
+1. On Node 1, bind the socket.
+
+   ```
+   > udp bind :: 1234
+   Done
+   ```
+
+1.  On Node 2, open the UDP socket.
```
Comment (jwhui) [inline site/en/reference/cli/concepts/udp.md:132]:
```suggestion
1. On Node 2, open the UDP socket.
```

File: site/en/reference/cli/concepts/udp.md [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+
+    ```
+    > udp open
+    Done
+    ```
+
+1.  On Node 2, use the `udp connect` command to open communication to Node 1.
```
Comment (jwhui) [inline site/en/reference/cli/concepts/udp.md:139]:
```suggestion
1. On Node 2, use the `udp connect` command to open communication to Node 1.
```

File: site/en/reference/cli/concepts/udp.md [hunk 16]
```diff
@@ -0,0 +1,186 @@
...
+    as the Node 2 source. The socket chooses an IP address and port randomly in this scenario.
+
+    For complete options with `udp send`, refer to
+    [udp send](https://openthread.io/reference/cli/commands#udp_send).
+
+1. Node 1 confirms receipt of the message from Node 2:
+
+   ```
+   > 5 bytes from fdde:ad00:beef:0:dac3:6792:e2e:90d8 49153 hello
+   ```
```
Comment (jwhui) [inline site/en/reference/cli/concepts/udp.md:109]:
```suggestion
    ```
    > 5 bytes from fdde:ad00:beef:0:dac3:6792:e2e:90d8 49153 hello
    ```
```

File: site/en/reference/cli/concepts/udp.md [hunk 17]
```diff
@@ -0,0 +1,186 @@
...
+    > udp open
+    Done
+    ```
+
+1. On Node 2, send a simple message to Node 1. 
+
+   ```
+   > udp send fdde:ad00:beef:0:bb1:ebd6:ad10:f33 1234 hello
+   Done
+   ```
```
Comment (jwhui) [inline site/en/reference/cli/concepts/udp.md:94]:
```suggestion
   ```
    > udp send fdde:ad00:beef:0:bb1:ebd6:ad10:f33 1234 hello
    Done
    ```
```

File: site/en/reference/cli/concepts/udp.md [hunk 18]
```diff
@@ -0,0 +1,186 @@
...
+    > udp bind :: 1234
+    Done
+    ```
+
+    The use of `::` denotes that the `bind` should use the unspecified IPv6 address,
+    thereby having the UDP/IPv6 stack assign the binding IPv6 address. For complete
+    options with `udp bind`, such as binding to a network interface, 
+    refer to [udp bind](https://openthread.io/reference/cli/commands#udp_bind).
+
+1.  On Node 2, open the UDP socket.
```
Comment (jwhui) [inline site/en/reference/cli/concepts/udp.md:82]:
```suggestion
1. On Node 2, open the UDP socket.
```

**Review by jwhui — COMMENTED** [review]
Content looks good. Suggested some minor fixes to make the indents consistent.

File: site/en/reference/cli/concepts/udp.md (hunk 18 above)
Comment (jrhodie) [reply site/en/reference/cli/concepts/udp.md:82]: thx Jonathan - making changes now...

**Review by jwhui — APPROVED** [review]

**Pull request merged**